src/LanguageServer/Impl/Implementation/Server.cs

```diff
@@ -374,7 +375,18 @@ private sealed class ReloadModulesQueueItem : IAnalyzable {
         private async Task DoInitializeAsync(InitializeParams @params, CancellationToken token) {
             _disposableBag.ThrowIfDisposed();
 
+            if (@params.rootUri != null) {
+                _rootDir = @params.rootUri.ToAbsolutePath();
+            } else if (!string.IsNullOrEmpty(@params.rootPath)) {
+                _rootDir = PathUtils.NormalizePath(@params.rootPath);
+            }
+
             Analyzer = await AnalysisQueue.ExecuteInQueueAsync(ct => CreateAnalyzer(@params.initializationOptions.interpreter, token), AnalysisPriority.High);
+            Version version = Version.Parse(@params.initializationOptions.interpreter.properties.TryGetValue("Version", out var o) ? o as string : null);
+            IndexingFileParser IndexingFileParser = new IndexingFileParser(SymbolIndex, version.ToLanguageVersion());
+            WorkspaceManager = new WorkspaceManager(_rootDir, @params.initializationOptions.includeFiles, @params.initializationOptions.excludeFiles, IndexingFileParser);
+            WorkspaceManager.StartParseRootDir();
```

Comment:
WorkspaceManager should be disposable/stoppable. Typically `Dispose` would cancel the work. You can then add the object to the `disposableBag`